compiler/src/dmd/attrib.d

```diff
@@ -232,6 +238,12 @@ extern (C++) class StorageClassDeclaration : AttribDeclaration
         this.stc = stc;
     }
 
+    extern (D) this(const ref Loc loc, StorageClass stc, Dsymbols* decl)
+    {
+        super(loc, decl);
+        this.stc = stc;
```

Comment:
This is a constructor. You can set the loc here and ditch the dsymbol and attrib constructors.

Reply:
That's what I initially tried, but this code
```
extern (D) this(const ref Loc loc, StorageClass stc, Dsymbols* decl)
    {
        super(decl);
        this.stc = stc;
        this.loc = loc;
    }
```
Does not work, it results in the following error: `Error: cannot modify `const` expression `this.loc``

Reply:
I guess you can use the [existing constructor](https://github.com/dlang/dmd/pull/14720/files#diff-81dc814caef1fd4471b91a93bfaae678436028ece215e0f9d624feb42a2b356cR70) in AttribDeclaration, but pass a null identifier.